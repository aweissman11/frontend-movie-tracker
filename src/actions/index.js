export const getUserLoggedIn = (id, name) => {
  return {
    type: 'GET_USER_LOGGED_IN',
    user: { id, name }
  }
}

export const logUserOut = () => {
  return {
    type:  'LOG_USER_OUT'
  }
}

export const displayLogin = () => {
  return {
    type: 'DISPLAY_LOGIN'
  }
}

export const displaySignUp = () => {
  return {
    type: 'DISPLAY_SIGN_UP'
  }
}

export const setFavorites = (favorites = []) => {
  return {
    type: 'SET_FAVORITES',
    favorites
  }
}

export const updateFilters = (filters) => {
  return {
    type: 'UPDATE_FILTERS',
    filters
  }
}
export const updateSearchQuery = (searchQuery) => {
  return {
    type: 'UPDATE_QUERY',
    searchQuery
  }
}

export const setMovieList = (movies) => {
  return {
    type: 'SET_MOVIE_LIST',
    movies
  }
}

export const setHasErrored = (bool) => {
  return {
    type: 'SET_HAS_ERRORED',
    hasErrored: bool
  }
}

export const setIsOk = (bool) => {
  return {
    type: 'SET_IS_NOT_OK',
    notOk: bool
  }
}

export const isLoading = (bool) => {
  return {
    type: 'IS_LOADING',
    isLoading: bool
  }
}