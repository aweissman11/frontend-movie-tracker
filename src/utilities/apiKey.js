export const apiKey = 'db393102c6d38bb71fed129f3aaf36d8';