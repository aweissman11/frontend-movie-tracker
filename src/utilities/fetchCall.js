const fetchCall = async (url) => {
  const response = await fetch(url);
  if (!response.ok) {
    console.error(Error(response.statusText));
    return 'failed'
  }
  return await response.json();
}

export default fetchCall;