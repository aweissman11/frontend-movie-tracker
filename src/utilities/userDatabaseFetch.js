export const checkUserList = async (data) => {
  const url = 'http://localhost:3000/api/users'

  const optionsObject = {
    method: "POST", 
    body: JSON.stringify(data),
    credentials: "same-origin", 
    headers: {
      "Content-Type": "application/json",
    }
  }

  const response = await fetch(url, optionsObject);
  return await response.json();
}

export const createNewUser = async (data) => {
  const url = 'http://localhost:3000/api/users/new'

  const optionsObject = {
    method: "POST", 
    body: JSON.stringify(data),
    credentials: "same-origin", 
    headers: {
        "Content-Type": "application/json",
    }
  }

  const response = await fetch(url, optionsObject);
  return await response.json();
}

export const addFavorite = async (data) => {
  const url = 'http://localhost:3000/api/users/favorites/new'

  const optionsObject = {
    method: "POST", 
    body: JSON.stringify(data),
    credentials: "same-origin", 
    headers: {
        "Content-Type": "application/json",
    }
  }

  const response = await fetch(url, optionsObject);
  return await response.json();
}

export const removeFavorite = async (userId, movieId) => {
  const url = `http://localhost:3000/api/users/${userId}/favorites/${movieId}`

  const optionsObject = {
    method: "DELETE", 
    body: JSON.stringify({
      user_id: userId,
      movie_id: movieId
    }),
    credentials: "same-origin", 
    headers: {
        "Content-Type": "application/json",
    }
  }

  const response = await fetch(url, optionsObject);
  return await response.json();
}

