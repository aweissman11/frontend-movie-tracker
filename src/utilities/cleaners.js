import { apiKey } from "./apiKey";

export const getMoviePosterUrl = (movies) => {
  return movies.map( movie => {
    return movie.poster_path
  })
}

export const getTodaysDate = () => {
  const today = new Date();
  let dd = today.getDate();
  let mm = today.getMonth()+1;
  let yyyy = today.getFullYear();
  if(dd<10) {
      dd = '0'+dd;
  } 
  if(mm<10) {
      mm = '0'+mm;
  } 
  return `${mm}-${dd}-${yyyy}`;
}

export const getFilterUrl = (filterProperties) => {
  const today = getTodaysDate();

  let url = `https://api.themoviedb.org/3/discover/movie?api_key=${apiKey}&language=en-US?primary_release_date.lte=${today}&sort_by=popularity.desc`

  const { genre, year, rating, sort } = filterProperties;
  if (genre && genre !== 'GENRE') {
    url += `&with_genres=${genre}`
  }
  
  if (year && year !== 'YEAR') {
    url += `&year=${year}`
  }

  if (rating && rating !== 'RATING') {
    url += `&certification_country=US&certification=${rating}`
  }

  if (sort && sort !== 'SORT') {
    url += `&sort_by=${sort}`
  }

  return url;

}