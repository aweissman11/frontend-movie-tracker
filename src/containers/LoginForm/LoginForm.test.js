import React from 'react';
import { shallow } from 'enzyme';
import { connect } from 'react-redux';
import * as userDatabaseFetch from '../../utilities/userDatabaseFetch';

import { LoginForm, mapDispatchToProps, mapStateToProps } from './index';
import * as Actions from '../../actions';

describe('LoginForm', () => {
  let wrapper;
  const mockUser = {
    id: 1
  }
  const mockShowLogin = true;
  const mockLogUserIn = jest.fn();
  const mockDisplaySignUp = jest.fn();
  const mockSubscribe = jest.fn();
  const mockDispatch = jest.fn();
  const mockGetState = jest.fn();
  const mockUserDatabaseFetch = {
    checkUserList: jest.fn().mockImplementation(() => {
      return Promise.resolve({
        data: {
          id: 1,
          name: 'mock name'
        }
      });
    })
  }

  const mockStore = {
    subscribe: mockSubscribe,
    dispatch: mockDispatch,
    getState: mockGetState
  };

  const defaultState = {
    email: '',
    password: '',
    loginError: '',
    userDatabaseFetch: userDatabaseFetch
  };

  const mockPreventDefault = jest.fn();

  const mockEvent = {
    target: {
      name: 'email',
      value: 'email@email.com'
    },
    preventDefault: mockPreventDefault
  }

  beforeEach(() => {
    wrapper = shallow(<LoginForm 
      store={mockStore}
      user={mockUser}
      showLogin={mockShowLogin}
      logUserIn={mockLogUserIn}
      displaySignUp={mockDisplaySignUp}
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    expect(wrapper.state()).toEqual(defaultState)
  });

  describe('handleChange', () => {
    it('should call set state', async () => {
      const expected = 'email@email.com'
      await wrapper.instance().handleChange(mockEvent);

      expect(wrapper.state().email).toEqual(expected);
    });
  });

  describe('removeWarning', () => {
    it('should call set state', async () => {
      const expected = '';

      await wrapper.instance().removeWarning();

      expect(wrapper.state().loginError).toEqual('');
    });
  })

  describe('userWarning', () => {
    it('should call set state', async () => {
      const mockType = 'loginError'
      const mockWarning = 'mock warning'

      await wrapper.instance().userWarning(mockType, mockWarning);

     expect(wrapper.state()[mockType]).toEqual(mockWarning);
    });

    // it('should call removeWarning', async () => {
    //   const mockRemoveWarning = jest.fn();

    //   wrapper.instance().removeWarning = mockRemoveWarning;
    //   await wrapper.instance().userWarning()

    //   expect(mockRemoveWarning).toHaveBeenCalled()
    // });
  });

  describe('submitLogin', () => {

    const mockCheckUserList = jest.fn().mockImplementation(() => {
      return Promise.resolve({
        data: {
          id: 1,
          name: 'mock name'
        }
      });
    });

    const mockUserDataBaseFetch = {
      checkUserList: mockCheckUserList
    }

    const mockEmail = 'mock email';
    const mockPassword = 'mock password';

    beforeEach(async () => {
      await wrapper.setState({
        email: mockEmail,
        password: mockPassword,
        userDatabaseFetch: mockUserDataBaseFetch
      });
    });

    it('should call preventDefault', async () => {
      await wrapper.instance().submitLogin(mockEvent);

      expect(mockPreventDefault).toHaveBeenCalled();
    });

    it('should call checkUserList with the correct params', async () => {
      const expected = {
        email: mockEmail,
        password: mockPassword,
      }
      await wrapper.instance().submitLogin(mockEvent);

      expect(mockCheckUserList).toHaveBeenCalledWith(expected);
    });

    it('should call logUserIn with the correct params', async () => {
      const expected = {data : {
        id: 1, name: 'mock name'
      }}

      expect(mockLogUserIn).toHaveBeenCalledWith(1, 'mock name');
    });

    it('should call userWarning on failed fetch', async () => {
      const mockFailedCall = jest.fn().mockImplementation(() => {
        return Promise.reject({
          error: 'error'
        });
      });

      const mockFailedDatabase = {checkUserList: mockFailedCall}
      const mockUserWarning = jest.fn();

      await wrapper.setState({
        userDatabaseFetch: mockFailedDatabase
      });

      wrapper.instance().userWarning = mockUserWarning

      await wrapper.instance().submitLogin(mockEvent);

      expect(mockUserWarning).toHaveBeenCalled();
    })

  });

});