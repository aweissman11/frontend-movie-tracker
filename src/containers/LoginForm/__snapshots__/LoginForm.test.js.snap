// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginForm
    displaySignUp={[MockFunction]}
    logUserIn={[MockFunction]}
    showLogin=""
    store={
      Object {
        "dispatch": undefined,
        "getState": undefined,
        "subscribe": undefined,
      }
    }
    user={
      Object {
        "id": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          <span>
            movie
          </span>
          Tracker
        </h1>,
        <h4>
          for the love of film
        </h4>,
        <div
          className="login-form-logo-wrapper"
        >
          <Logo />
        </div>,
        <input
          className="email-input"
          name="email"
          onChange={[Function]}
          placeholder="email"
          value=""
        />,
        <img
          alt="mail"
          className="email-icon"
          src="./email.png"
        />,
        <input
          className="password-input"
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="password"
          value=""
        />,
        <img
          alt="lock"
          className="password-icon"
          src="./password.png"
        />,
        <input
          className="login-submit"
          type="submit"
          value="login"
        />,
        <button
          className="sign-up-button"
          name="sign-up-button"
          onClick={[Function]}
        >
          signup
        </button>,
        <section
          className="login-social-wrapper"
        >
          <i
            className="fab fa-facebook login-social"
          />
          <i
            className="fab fa-twitter login-social"
          />
        </section>,
        <Link
          replace={false}
          to="/movies"
        >
          <button
            className="skip-login"
          >
            skip login
          </button>
        </Link>,
        <div
          className="login-error-wrapper "
        >
          <p
            className="login-error-text"
          >
            incorrect email/password combination
          </p>
        </div>,
      ],
      "className": "login-form ",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              movie
            </span>,
            "Tracker",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "movie",
            },
            "ref": null,
            "rendered": "movie",
            "type": "span",
          },
          "Tracker",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "for the love of film",
        },
        "ref": null,
        "rendered": "for the love of film",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Logo />,
          "className": "login-form-logo-wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "email-input",
          "name": "email",
          "onChange": [Function],
          "placeholder": "email",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "mail",
          "className": "email-icon",
          "src": "./email.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "password-input",
          "name": "password",
          "onChange": [Function],
          "placeholder": "password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "lock",
          "className": "password-icon",
          "src": "./password.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "login-submit",
          "type": "submit",
          "value": "login",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "signup",
          "className": "sign-up-button",
          "name": "sign-up-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "signup",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="fab fa-facebook login-social"
            />,
            <i
              className="fab fa-twitter login-social"
            />,
          ],
          "className": "login-social-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fab fa-facebook login-social",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fab fa-twitter login-social",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="skip-login"
          >
            skip login
          </button>,
          "replace": false,
          "to": "/movies",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "skip login",
            "className": "skip-login",
          },
          "ref": null,
          "rendered": "skip login",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p
            className="login-error-text"
          >
            incorrect email/password combination
          </p>,
          "className": "login-error-wrapper ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "incorrect email/password combination",
            "className": "login-error-text",
          },
          "ref": null,
          "rendered": "incorrect email/password combination",
          "type": "p",
        },
        "type": "div",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            <span>
              movie
            </span>
            Tracker
          </h1>,
          <h4>
            for the love of film
          </h4>,
          <div
            className="login-form-logo-wrapper"
          >
            <Logo />
          </div>,
          <input
            className="email-input"
            name="email"
            onChange={[Function]}
            placeholder="email"
            value=""
          />,
          <img
            alt="mail"
            className="email-icon"
            src="./email.png"
          />,
          <input
            className="password-input"
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
          />,
          <img
            alt="lock"
            className="password-icon"
            src="./password.png"
          />,
          <input
            className="login-submit"
            type="submit"
            value="login"
          />,
          <button
            className="sign-up-button"
            name="sign-up-button"
            onClick={[Function]}
          >
            signup
          </button>,
          <section
            className="login-social-wrapper"
          >
            <i
              className="fab fa-facebook login-social"
            />
            <i
              className="fab fa-twitter login-social"
            />
          </section>,
          <Link
            replace={false}
            to="/movies"
          >
            <button
              className="skip-login"
            >
              skip login
            </button>
          </Link>,
          <div
            className="login-error-wrapper "
          >
            <p
              className="login-error-text"
            >
              incorrect email/password combination
            </p>
          </div>,
        ],
        "className": "login-form ",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                movie
              </span>,
              "Tracker",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "movie",
              },
              "ref": null,
              "rendered": "movie",
              "type": "span",
            },
            "Tracker",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "for the love of film",
          },
          "ref": null,
          "rendered": "for the love of film",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Logo />,
            "className": "login-form-logo-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "email-input",
            "name": "email",
            "onChange": [Function],
            "placeholder": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "mail",
            "className": "email-icon",
            "src": "./email.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "password-input",
            "name": "password",
            "onChange": [Function],
            "placeholder": "password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "lock",
            "className": "password-icon",
            "src": "./password.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "login-submit",
            "type": "submit",
            "value": "login",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "signup",
            "className": "sign-up-button",
            "name": "sign-up-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "signup",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fab fa-facebook login-social"
              />,
              <i
                className="fab fa-twitter login-social"
              />,
            ],
            "className": "login-social-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fab fa-facebook login-social",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fab fa-twitter login-social",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="skip-login"
            >
              skip login
            </button>,
            "replace": false,
            "to": "/movies",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "skip login",
              "className": "skip-login",
            },
            "ref": null,
            "rendered": "skip login",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p
              className="login-error-text"
            >
              incorrect email/password combination
            </p>,
            "className": "login-error-wrapper ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "incorrect email/password combination",
              "className": "login-error-text",
            },
            "ref": null,
            "rendered": "incorrect email/password combination",
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
